### PR TITLE
Improve AssetSelect and refactor Switch decorator

### PR DESCRIPTION
**Pull request by dfguerrerom**
AssetSelect:
- Define `default_asset` parameter as trait in `SelectAsset` widget and have the chance to update the default assets once the widget is instantiated.

Switch decorator:
- Extend the behavior as mentioned in https://github.com/12rambau/sepal_ui/issues/263#issuecomment-912615658, It would be really nice if you can check it.

### CODE REVIEW

**Review by 12rambau — CHANGES_REQUESTED** [review]

**Comment by 12rambau** [comment]
Thanks a lot for the PR. I really appreciater the switch decorator I'm positive it will be useful in lots of situations. I finally wrapped my head around the assetSelect trait and I effectively agree on the use of a `List`trait. 

I made super small modifications and suggested some that would be nice to see before merging in master

File: sepal_ui/scripts/utils.py [hunk 1]
```diff
@@ -335,31 +335,54 @@ def to_colors(in_color, out_type="hex"):
...
+            if on_widgets:
+
+                def w_assign(value):
+                    if not all([hasattr(self, w_name) for w_name in on_widgets]):
+                        raise Exception(
+                            "The provided widget(s) does not exist in the current class"
+                        )
+
+                    for w_name in on_widgets:
+                        widget = eval(f"self.{w_name}")
```
[the remark this answers is not in the document]
Comment (dfguerrerom) [reply sepal_ui/scripts/utils.py:365]: of course!

File: sepal_ui/scripts/utils.py [hunk 2]
```diff
@@ -335,31 +335,54 @@ def to_colors(in_color, out_type="hex"):
...
-def switch(*params, debug=True):
+
+def switch(*params, debug=True, on_widgets=[]):
     """
-    Decorator to switch the state of input boolean parameters.
-    
+    Decorator to switch the state of input boolean parameters on class widgets or the
+    class itself. If on_widgets are defined, it will switch the state of every widget
+    parameter, otherwise it will change the state of the class (self). You can also set
+    two decorators on the same function, one for
```
[the remark this answers is not in the document]
Comment (dfguerrerom) [reply sepal_ui/scripts/utils.py]: solved 1edcfd142086cb0a5bb7baf95006e0b3f6bacdb3

File: sepal_ui/scripts/utils.py [hunk 3]
```diff
@@ -335,31 +335,54 @@ def to_colors(in_color, out_type="hex"):
...
     def decorator_switch(func):
         @wraps(func)
         def wrapper_switch(self, *args, **kwargs):
-            [setattr(self, param, True) for param in params]
-            
+
+            if on_widgets:
+
+                def w_assign(value):
+                    if not all([hasattr(self, w_name) for w_name in on_widgets]):
```
[the remark this answers is not in the document]
Comment (dfguerrerom) [reply sepal_ui/scripts/utils.py]: 1edcfd142086cb0a5bb7baf95006e0b3f6bacdb3

File: sepal_ui/scripts/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dfguerrerom) [reply sepal_ui/scripts/utils.py]: 1edcfd142086cb0a5bb7baf95006e0b3f6bacdb3

**Comment by dfguerrerom** [comment]
> Thanks a lot for the PR. I really appreciater the switch decorator I'm positive it will be useful in lots of situations. I finally wrapped my head around the assetSelect trait and I effectively agree on the use of a `List`trait.
> 
> I made super small modifications and suggested some that would be nice to see before merging in master

I used `Any` trait since we are using both string and list types as inputs (to avoid crash the apps that are currently using the string)

**Review by 12rambau — APPROVED** [review]

**Comment by dfguerrerom** [comment]
@12rambau I think that is better if we review the reclassify branch before pass the black format to everything. We could end with lots of conflicts.

**Comment by 12rambau** [comment]
> @12rambau I think that is better if we review the reclassify branch before pass the black format to everything. We could end with lots of conflicts.

oups 1s too late

**Pull request merged**